apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1035,6 +1035,25 @@ function validateAssessment({
       errors.push('Real-time grading cannot be disabled for Homework-type assessments');
     }
 
+    // Check that real-time grading is not disabled at any level in the hierarchy
+    (assessment.zones || []).forEach((zone) => {
+      if (zone.allowRealTimeGrading === false) {
+        errors.push('Real-time grading cannot be disabled for Homework-type assessments');
+      }
+
+      (zone.questions || []).forEach((question) => {
+        if (question.allowRealTimeGrading === false) {
+          errors.push('Real-time grading cannot be disabled for Homework-type assessments');
+        }
+
+        (question.alternatives || []).forEach((alternative) => {
```

Comment:
nit: Should you do a `.some` here instead of `forEach`

Reply:
Addressed in c14d4b5.